src/lib.rs

```diff
@@ -527,16 +533,137 @@ impl LightningNode {
         }
     }
 
-    // TODO remove unused_variables after breez sdk implementation
-    #[allow(unused_variables)]
-    pub fn get_latest_payments(&self, number_of_payments: u32) -> Result<Vec<Payment>> {
-        todo!()
+    pub fn get_latest_payments(&self, _number_of_payments: u32) -> Result<Vec<Payment>> {
+        let breez_payments = self
+            .rt
+            .handle()
+            .block_on(self.sdk.list_payments(PaymentTypeFilter::All, None, None))
+            .map_to_runtime_error(
+                RuntimeErrorCode::NodeUnavailable,
+                "Failed to get payment by hash",
+            )?;
+
+        let mut payments = Vec::new();
+
+        for breez_payment in breez_payments {
+            if breez_payment.payment_type == breez_sdk_core::PaymentType::ClosedChannel {
+                continue;
+            }
+            let payment = self.payment_from_breez_payment(breez_payment)?;
+            payments.push(payment);
+        }
+
+        Ok(payments)
     }
 
-    // TODO remove unused_variables after breez sdk implementation
-    #[allow(unused_variables)]
     pub fn get_payment(&self, hash: String) -> Result<Payment> {
-        todo!()
+        let breez_payment = self
+            .rt
+            .handle()
+            .block_on(self.sdk.payment_by_hash(hash))
+            .map_to_runtime_error(
+                RuntimeErrorCode::NodeUnavailable,
+                "Failed to get payment by hash",
+            )?
+            .ok_or_invalid_input("Invalid hash: no payment with provided hash was found")?;
+
+        self.payment_from_breez_payment(breez_payment)
+    }
+
+    fn payment_from_breez_payment(
+        &self,
+        breez_payment: breez_sdk_core::Payment,
+    ) -> Result<Payment> {
+        let payment_details = if let PaymentDetails::Ln { data } = breez_payment.details {
+            data
+        } else {
+            return Err(permanent_failure(
+                "Current interface doesn't support PaymentDetails::ClosedChannel",
+            ));
+        };
```

Comment:
```suggestion
        match breez_payment.details {
            PaymentDetails::Ln { data } => data,
            _ => return Err(permanent_failure("Current interface doesn't support PaymentDetails::ClosedChannel")),
        }
```

Reply:
Applied in [ce72943](https://github.com/getlipa/lipa-lightning-lib/pull/637/commits/ce7294377591216ade8018100c4a24b3b1118322)